### PR TITLE
⬆️ Update foundry-gas-diff

### PR DESCRIPTION
**Pull request by Rubilmax**

### CODE REVIEW

File: .github/workflows/foundry-gas-diff.yml [hunk 1]
```diff
@@ -28,13 +28,16 @@ jobs:
           version: nightly
 
       - name: Run tests
-        run: forge test --gas-report > gasreport.ansi
+        run: forge test --gas-report | tee gasreport.ansi
+        env:
+          # make fuzzing semi-deterministic to avoid noisy gas cost estimation
+          # due to non-deterministic fuzzing, but keep it not always deterministic
+          FOUNDRY_FUZZ_SEED: 0x${{ github.event.pull_request.base.sha || github.sha }}
```
Comment (MathisGD) [inline .github/workflows/foundry-gas-diff.yml:35]: Idk if this is an issue in this repo 🤔

Comment (Rubilmax) [reply .github/workflows/foundry-gas-diff.yml:35]: It's not an issue currenlty but still, we expect the CI to be reproducible

Comment (MathisGD) [reply .github/workflows/foundry-gas-diff.yml:35]: OK but imagine this seed does not test a edge case but an other do, we could miss it

Comment (MerlinEgalite) [reply .github/workflows/foundry-gas-diff.yml:35]: Yes agree with the last argument

Comment (MerlinEgalite) [reply .github/workflows/foundry-gas-diff.yml:35]: Although the CI seems to fail [here](https://github.com/morpho-dao/morpho-utils/pull/46).

Comment (Rubilmax) [reply .github/workflows/foundry-gas-diff.yml:35]: We'd miss it on this PR, but not on another or when the CI runs on dev/main (hence the goal of keeping it non-deterministic)...

**Review by MerlinEgalite — APPROVED** [review]
LGTM

File: .github/workflows/foundry-gas-diff.yml (hunk 1 above)
Comment (MathisGD) [reply .github/workflows/foundry-gas-diff.yml:35]: so we keep the deterministic CI ?
